### PR TITLE
Optimize testing for int subtype that is a single range

### LINKED ISSUE #890: Optimize testing for int subtype that is a single range
Special case testing an int representation against an int subtype that is a single range.

### PR DESCRIPTION
**Pull request by ushirask**
Fixes #890

### CODE REVIEW

**Review by jclark — CHANGES_REQUESTED** [review]
Looks good but we need some more tests.

- more ranges e.g. int:Signed32
- a range with gaps in it (i.e. `all` not true)
- type cast

File: compiler/modules/nback/typeTest.bal [hunk 1]
```diff
@@ -94,7 +94,15 @@ function buildTypeTestedValue(llvm:Builder builder, Scaffold scaffold, bir:Regis
         }
     }
     else if baseRepr == BASE_REPR_INT {
-        hasType = <llvm:Value>builder.call(scaffold.getRuntimeFunctionDecl(typeContainsIntFunction), [scaffold.getTypeTest(<t:ComplexSemType>semType), value]);
+        t:IntSubtypeConstraints? intConstraints = t:intSubtypeConstraints(semType);
+        if intConstraints != () && intConstraints.all {
+            llvm:Value hasType1 = builder.iCmp("sle", llvm:constInt(LLVM_INT, intConstraints.min), value);
+            llvm:Value hasType2 = builder.iCmp("sge", llvm:constInt(LLVM_INT, intConstraints.max), value);
```
Comment (manuranga) [inline compiler/modules/nback/typeTest.bal]: instead hasType1 and hasType2, how about `lowerBoundMet` and `upperBoundMet`.

Comment (jclark) [reply compiler/modules/nback/typeTest.bal]: Do we really need the temp variables?

Comment (ushirask) [reply compiler/modules/nback/typeTest.bal]: I'll remove the temp variables

**Pull request merged**